### ISSUE TITLE
Banner Advertisements No Longer Showing

### ISSUE THREAD

**Issue by Yaanu** [ccd0/4chan-x#975]
To show my support for 4chan, as well as to laugh at people who buy ad-space, I've left banner advertisements visible despite having the opportunity to remove them. However, recently I've found the banner ads to have disappeared completely. I've isolated the problem to 4chan X; OneeChan has the ability to remove ads, but isolating that extension keeps the ads as intended.

I'm not sure why this is the case, but if I had to guess, it may have something to do with the image host for the banner ads being blacklisted by the extension, though I'm not even sure if 4chan X has that capability.

OS/Browser Specifications:
- OS X El Capitan (Version 10.11.5)
- Chrome Version 51.0.2704.103 (64-bit)
- Tampermonkey v4.0.69
- 4chan X v1.11.35.6
- OneeChan v5.5.39

**Comment by ccd0** [comment]
After 4chan was sold to Hiroyuki, he began running ads from malware distributors. If 4chan X left ads unblocked by default, users would be randomly redirected to other websites, and some users would be infected with malware. 4chan X has attempted to block the nastier ad servers for some time (although it's not perfect especially with Tampermonkey and using an ad blocker is still recommended), but Hiroyuki continuously adds new ones, and introduces nasty ads in places they weren't before. So 4chan X has been switched to a whitelist approach. If you want the scripts from a particular ad server to run, you can identify what's being blocked by "Content Security Policy" messages in the console (F12), and add them to the `Javascript Whitelist` on the `Advanced` setting tab. But he has used all of the ad servers for malware, even the ones serving the old banner ads, so I wouldn't recommend running scripts from any of them:

![banner ad](https://cloud.githubusercontent.com/assets/5894749/16250322/5156c178-37d1-11e6-8ff1-668ca33e8538.png)

**Comment by Yaanu** [comment]
I must have missed that in the changelog, then. Thank you for the explanation.

Personally, I've never had any problems with the advertisements and malware, though I was aware of the problem when it happened. It may be because I don't browse many boards other than /co/, but I can't say for sure.

Again, thank you. This does make me wonder how the banners that will be advertising other boards in [the new banner contest](https://www.4chan.org/banner-contest) will be handled, since they won't necessarily need to be handled externally.

**Comment by ccd0** [comment]
It's a creeping problem. Hiroyuki seems to be trying to boil the proverbial frog.

Once the banners from the contest are up, if they end up being loaded with the ad server's Javascript, I could probably add an option to load them with 4chan X.

**Comment by i-cant-git** [comment]
@ccd0 Do you think he wants to botnet all 4chan users?

**Comment by ccd0** [comment]
I think he wants to make money and is indifferent to users' computers becoming part of botnets.

**Comment by ccd0** [comment]
Opened a new issue for the contest thing: #981
